### ISSUE TITLE
Any plans to support Cryptonight-R

### ISSUE THREAD

**Issue by BKdilse** [fireice-uk/xmr-stak#2254]
Hi,

Are you looking at implementing support for the Cryptonight-R Algo, which Monero are forking to in March?

I Know XMRig have added in the changes, but I prefer XMR-Stak :)

**Comment by fireice-uk** [comment]
Yes, we are working on it

**Comment by BKdilse** [comment]
Thanks, appreciated .

**Comment by cryptatometrader44** [comment]
@BKdilse You did well to ask this question because I had asked it to me for a few days as well.

I also prefer this minning software to XMrig

@ Freice-UK

Thank you for your work to you and psychocrypt.
and
thanks also for this information and quick response.

**Comment by BKdilse** [comment]
Yeah, fireice has had some heat recently, but I dont believe in that shit.  Good work guys.  Always willing to help where I can.

**Comment by jwatte** [comment]
The fork is drawing closer...

**Comment by fireice-uk** [comment]
Yes, we won't get it done much before due to the emergency nature of the fork. But it should be out next week.

**Comment by psychocrypt** [comment]
We will bring a beta veraion out as soon as possible. we are currently
implementing the last stuff and than it must pass our own tests first.

**Comment by valiant1x** [comment]
FYI @psychocrypt current feature-cryptonightR branch is not passing hash tests. Pool rejects all shares as low difficulty and the POW hash produced by xmr-stak fails validation in both pool and daemon tests:

The hash tests pass properly for the [hashes provided by Monero](https://github.com/monero-project/monero/blob/release-v0.13/tests/hash/tests-slow-4.txt). For example, for the given blob, xmr-stak hashed `80365a1c919c2b5b978b9973fc2460d3f3d65e5eb3919c68c39142d394a12600` while the pool and daemon hashed `6d8fca7873dacd09bc51fbf022466c3b33280220afd74bab0f34140f18d6eb6f`.

```
9/14 Test: hash-slow-4
Output:
----------------------------------------------------------
Hash mismatch on test 11
Input: 0606c4efe5e3052607db63efcab1b3bb6568cd7602718f3e9d91f5d32504c4dd8257807f618ccfb5110000013abb4f71ab44391536b2787eb94e5a86cb021806a5dac727d7b66542c030c501
Expected hash: 80365a1c919c2b5b978b9973fc2460d3f3d65e5eb3919c68c39142d394a12600
Actual hash: 6d8fca7873dacd09bc51fbf022466c3b33280220afd74bab0f34140f18d6eb6f
<end of output>
Test time =   1.32 sec
```
Test:
`80365a1c919c2b5b978b9973fc2460d3f3d65e5eb3919c68c39142d394a12600 0606c4efe5e3052607db63efcab1b3bb6568cd7602718f3e9d91f5d32504c4dd8257807f618ccfb5110000013abb4f71ab44391536b2787eb94e5a86cb021806a5dac727d7b66542c030c501 801`

Of course, pool rejects all of these shares as low difficulty since they are not hashing correctly on pool side.

**Comment by psychocrypt** [comment]
This branch is not up to date. I have all still on my system and must fix a
few amd issues first before I will give out a beta
